docs/logging/README.md

```diff
@@ -0,0 +1,130 @@
+# Logging stack
+
+### Why do we need logging stack?
+We need logging stack because currently Kubernetes use the underlying container runtime logging which does not persist log for not existing containers. This makes difficult to investigate issues ocurred in long ago destroyed containers.
+Our logging stack is created to keep logs for 14 days. Also there are pretty UI which can be used for easier log filtering.
+
+
+### Architecture:
+* Fluent-bit Deamonset which works like a log collector. Also we use custom Golang plugin which spreads log messages to their Loki instances
+* One Loki Statefulset in the `garden` namespace which contains logs for the seed cluster and one per shoot namespace which contains logs for shoot's controlplane
+* One Grafana Deployment in `garden` namespace and two Deployments per shoot namespace(one exposed to the end users and one for the operators). Grafana is the UI component used in the logging stack.
+
+### How to access the logs
+We need to authenticate first in front of the Grafana ingress (Can be found in the Garden Dashboard in `Monitoring and Logging` section)
+After that there are two options for monitoring logs.
+* The first one is to log in in the admin panel (bottom left corner). The default username and password are `admin`, `admin`. Now we can choose `Explore` menu (Left side of the screen) and after that we can create custom filters based on the log labels supported in Loki and their values. For example: 
+```{pod_name='prometheus-0'}```
+or with regex:
+```{pod_name=~'prometheus.+'}```
+
+* The other option is to go to the `Dashboards` panel. As mentioned before, we have a custom dashboards for pod logs. There we have one selector field for `pod_name` and one search field where we can search for particular string in the log messages. The following dashboards can be used for logs:
+
+  * Garden Grafana
+    * Pod Logs
+    * Extensions
+    * Systemd Logs
+  * User Grafana
+    * Kubernetes Control Plane Status
+  * Operator Grafana 
+    * Kubernetes Pods
+    * Kubernetes Control Plane Status
+
+### Configuration
+#### Fluent-bit
+
+We can modify the fluent-bit configuration from `charts/seed-bootstrap/charts/fluent-bit/templates/fluent-bit-configmap.yaml`
+We can see that there are five different specifications
+* SERVICE: Where is defined the server specifications
+* INPUT: Where is defined the input stream of the logs
+* OUTPUT: Where is defined the output source (Loki for example)
+* FILTER: Where we filter logs with specific key
+* PARSER: Which is used by filters for parsing the log message
+
+All of the components have custom filters and parsers defined for parsing their log messages correctly. Currently in the extensions we create configmap under /charts directory called `logging-config` where their filters and parsers are. The fluent-bit reads this type of config files.
```

Comment:
Don't talk about the extensions. You can reference the extensions contract in docs/extensions if you wish.